helpers/Physics.h

```diff
@@ -0,0 +1,320 @@
+#ifndef _HELPERS_PHYSICS_H_
+#define _HELPERS_PHYSICS_H_
+
+#ifndef Z64HDR_HELPER_QUALIFIERS
+#define Z64HDR_HELPER_QUALIFIERS
+#endif
+
+#include "include/z64.h"
+#include "Matrix.h"
+
+typedef void(*PhysicCallback)(s32 limbIndex, void*, void*);
+
+typedef struct {
+	Vec3f pos;
+	Vec3f rot;
+	Vec3f vel;
+} PhysicsLimb;
+
+typedef struct {
+	s32 numLimbs;
+	f32 gravity;
+	f32 floorY;  // world.pos.y, won't go through this
+	f32 maxVel;  // Clamps velocity value
+	f32 velStep; // Values below 1.0f will give it spring like motion
+	f32 velMult; // Control the power of velocity
+} PhysicsInfo;
+
+typedef struct {
+	Vec3f pos;
+	Vec3s rot;
+	MtxF* mtxF;
+} PhysicsHead;
+
+typedef struct {
+	u32   dlist;
+	Vec3f scale;  // Gfx scale
+	u8    segID;  // For matrix
+	u8    noDraw; /* If there needs to be calculations
+	                 to get in position before drawing */
+} PhysicsGfx;
+
+typedef struct {
+	s32    num; // amount of spheres
+	Vec3f* centers;
+	f32    radius;
+} PhysicsSpheres; // "collision" spheres, pushes limbs away
+
+typedef struct {
+	u8    lockRoot;    // Prevent physics rotating root limb
+	Vec2f rotStepCalc; // DEG, limits rot to next limb in main calc
+	Vec2f rotStepDraw; // DEG, limits rot on draw, smoothens output
+} PhysicsConstraint;
+
+typedef struct {
+	s32   num;  // How many limbs will be smoothed with push
+	Vec3f push; // direction Z, pushes based on rot[0]
+	f32   mult; // How much the pushing fill affect
+	Vec3f rot;  // DEG, rigids towards, relative rot
+} PhysicsRigidity;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32 limbsLength[];
+} PhysicsStrandInit;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32* limbsLength;
+} PhysicsStrand;
+
+Z64HDR_HELPER_QUALIFIERS
+void Physics_GetHeadProperties(PhysicsStrand* strand, Vec3f* headPosModel) {
+	static MtxF mtxF;
+	Vec3f zero = { 0 };
+	
+	Matrix_Get(&mtxF);
+	strand->head.mtxF = &mtxF;
```

Comment:
this looks pretty bad, using a static MtxF means it's shared between all instances of an actor (all actors using the same overlay)
It isn't an issue with single actors (eg link)
and isn't actually an issue in general because `Physics_DrawDynamicStrand` is expected to run right after, but still...

I would suggest putting space for the matrix in PhysicsHead like `MtxF head;` (not a pointer)
```suggestion
	Vec3f zero = { 0 };
	
	Matrix_Get(&strand->head.mtxF);
```